### PR TITLE
sqlccl: test backup and restore of system tables

### PR DESCRIPTION
**Pull request by benesch**
@danhhz thinks there may be bugs lurking. From Slack, back in the day, ID 5 (zones) would hit internal keys that would cause MVCC iteration to fail.

### CODE REVIEW

File: pkg/ccl/backupccl/backup_test.go [hunk 1]
```diff
@@ -372,6 +372,51 @@ func backupAndRestore(
...
+
+	expectedFingerprints := make(map[string][][]string)
+	for _, table := range tables {
+		expectedFingerprints[table] = sqlDB.QueryStr(t,
+			fmt.Sprintf("SHOW EXPERIMENTAL_FINGERPRINTS FROM TABLE system.%s", table),
+		)
+	}
+
+	sqlDB.Exec(t, `BACKUP DATABASE system TO $1`, localFoo)
+	sqlDB.Exec(t, `RESTORE system.* FROM $1 WITH OPTIONS ('into_db'='system_new')`, localFoo)
```
Comment (dt) [inline pkg/ccl/backupccl/backup_test.go]: any reason for the old options syntax?

Comment (dt) [reply pkg/ccl/backupccl/backup_test.go]: oh, because this was a couple weeks before that merged.

Comment (benesch) [reply pkg/ccl/backupccl/backup_test.go]: Fixed.

**Comment by benesch** [comment]
omg, it's failing because https://github.com/cockroachdb/cockroach/pull/22440/files hasn't landed. Sigh.

**Comment by benesch** [comment]
Ok, PTAL! The trick was to take the backups and the fingerprints at the same system time (doh!), but this was held up by #22440.

**Review by dt — APPROVED** [review]

**Pull request merged**